sync3/lists.go

```diff
@@ -131,6 +143,7 @@ func (s *InternalRequestLists) RemoveRoom(roomID string) {
 
 func (s *InternalRequestLists) DeleteList(listKey string) {
 	// TODO
+	// TODO: could remove listKey's LastInterestedMessageTimestamp entries in allRooms
```

Comment:
I don't expect most clients will be repeatedly changing their list keys throughout the lifetime of one connection, but a malicious one could. That would eat up RAM. I think we probably ought to do this cleanup.

Reply:
My concern here isn't having one conn with thousands of simultaneous lists; it's a conn that

- creates list `a`
- next request, deletes list `a` and creates list `b`
- next request, deletes list `b` and creates list `c`
- next request, deletes list `c` and creates list `d`

etc

Reply:
Good point.